sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
+
+
+select city from customers where snum=(select snum from salespeople where sname="Serres");
++---------+
+| city    |
++---------+
+| Sanjose |
+| Berlin  |
++---------+
+
+
+
+
+23) Select all customers with a rating above 200. 
+
+
+select * from customers where rating>200;
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+
+
+
+24) Count the number of salespeople currently having orders in the orders table. 
+
+
+select count(distinct(snum)) as "No of saleseperson with orders" from customers;
++--------------------------------+
+| No of saleseperson with orders |
++--------------------------------+
+|                              5 |
++--------------------------------+
+
+
+
+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name, salesperson’s name and the salesperson’s rate of commission. 
+
+
+select c.cname, s.sname, s.comm from salespeople s inner join customers c where s.snum= c.snum and s.comm>12;
++----------+--------+------+
+| cname    | sname  | comm |
++----------+--------+------+
+| Liu      | Serres |   13 |
+| Grass    | Serres |   13 |
+| Cisneros | Rofkin |   15 |
++----------+--------+------+
+
+
+
+
+26) Find salespeople who have multiple customers. 
+
+select s.sname, count(c.cnum) from salespeople s inner join customers c on s.snum= c.snum group by c.snum having (count(c.cnum)>1);
+
++--------+---------------+
+| sname  | count(c.cnum) |
++--------+---------------+
+| Peel   |             2 |
+| Serres |             2 |
++--------+---------------+
+
+
+
+
+27) Find salespeople with customers located in their own cities. 
+
+select distinct(s.sname) from salespeople as s inner join customers as c on s.city= c.city;
++--------+
+| sname  |
++--------+
+| Fran   |
+| Motika |
+| Peel   |
+| Serres |
++--------+
+
+
+
+
+28) Find all salespeople whose name starts with ‘P’ and fourth character is ‘I’. 
+
+
+select sname from salespeople where sname like "P__I" ;
+Empty set (0.0007 sec)
+
+
+
+
+29) Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number. 
+
+
+select * from orders where cnum in (select cnum from customers where cname="Cisneros");
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-03-10 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-03-10 | 2008 | 1007 |
++------+---------+------------+------+------+
+
+
+
+
+
+30) Find the largest orders for Serres and Rifkin. 
+
+select s.sname, max(o.amt) from salespeople s inner join orders o on s.snum= o.snum group by sname having(s.sname="Serres" or s.sname="Rofkin");
+
```

Comment:
Having is used with Aggregated functions. Please use WHERE here.
Also, the query works fine with "GROUP BY sname" but it would be better if we use snum in place of sname.